src/ImageIO/CGImageDestination.cs

```diff
@@ -470,5 +495,15 @@ public bool CopyImageSource (CGImageSource image, CGImageDestinationOptions opti
 					o.Dispose ();
 			}
 		}
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[DllImport (Constants.ImageIOLibrary)]
+		static extern void CGImageDestinationAddAuxiliaryDataInfo (IntPtr /* CGImageDestinationRef* */ idst, IntPtr /* CFStringRef* */ auxiliaryImageDataType, IntPtr /* CFDictionaryRef* */ auxiliaryDataInfoDictionary);
+
+		public void AddAuxiliaryDataInfo (CGImageDestination dest, CGImageAuxiliaryDataType auxiliaryImageDataType, CGImageAuxiliaryDataInfo auxiliaryDataInfo)
+		{
+			var dict = auxiliaryDataInfo == null ? null : auxiliaryDataInfo.ToDictionary ();
+			CGImageDestinationAddAuxiliaryDataInfo (dest.Handle, auxiliaryImageDataType.GetConstant ().Handle, dict == null ? IntPtr.Zero : dict.Handle);
```

Comment:
1.  `dest` can be `null` and this would result in an `NRE`

2. we have an help to avoid `?:`, it's `NativeObjectHelper.GetHandle` (in src/ObjCRuntime/INativeObject.cs). It can ease readability.